### PR TITLE
Add utility classes for marshalling Wire Messages into HTTP requests

### PR DESCRIPTION
**Pull request by eeSeeGee**

### CODE REVIEW

File: misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import javax.inject.Inject
+
+/**
+ * This helper class is for making HTTP requests to an endpoint that handles protobuf Message claases.
+ */
+class ProtoMessageHttpClient {
+  @Inject
```
Comment (mmihic) [inline misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt]: nit: inject this as a ``val`` in the constructor

File: misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package misk.wire
```
Comment (mmihic) [inline misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt]: nit: move to `misk.client`

File: misk/src/main/kotlin/misk/wire/AbstractClientCommand.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package misk.wire
```
Comment (mmihic) [inline misk/src/main/kotlin/misk/wire/AbstractClientCommand.kt]: I'm not sure this is sufficient value-add to add at the misk-level - the only thing it's really doing is defining a parameter for passing in the URL. I think that should just be left to the applications.

File: misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        .addHeader("Accept", MediaTypes.APPLICATION_JSON)
+        .post(RequestBody.create(MediaTypes.APPLICATION_JSON_MEDIA_TYPE, requestJson))
+        .build()
+    val response = okHttp.newCall(request).execute()
+    if (response.code() != 200) {
+      throw RuntimeException("request failed (${response.code()} ${response.body()?.string()}"
+      )
+    }
+
+    return response.body()?.string()?.let { moshi.adapter(responseType).fromJson(it) }
```
Comment (mmihic) [inline misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt]: Please add tests.

File: misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt (hunk 2 above)
Comment (eeSeeGee) [reply misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt]: Sure

File: misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt (hunk 1 above)
Comment (eeSeeGee) [reply misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt]: Sure, makes sense.

File: misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt (hunk 4 above)
Comment (eeSeeGee) [reply misk/src/main/kotlin/misk/wire/ProtoMessageHttpClient.kt]: Whoops, will do.

File: misk/src/main/kotlin/misk/client/ProtoMessageHttpClient.kt [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import javax.inject.Inject
+
+/**
+ * This helper class is for making HTTP requests to an endpoint that handles protobuf Message claases.
+ */
+class ProtoMessageHttpClient @Inject constructor(var moshi: Moshi) {
+
+  private val okHttp = OkHttpClient()
+
+  fun <O : Any> post(
```
Comment (ryanhall07) [inline misk/src/main/kotlin/misk/client/ProtoMessageHttpClient.kt]: should this be private? would everybody use the friendlier version below?

**Comment by ryanhall07** [comment]
A couple of nit fixes I noticed reviewing after merge

https://github.com/square/misk/pull/106

File: misk/src/main/kotlin/misk/client/ProtoMessageHttpClient.kt (hunk 5 above)
Comment (mmihic) [reply misk/src/main/kotlin/misk/client/ProtoMessageHttpClient.kt]: can't make this private if we intend to inline it, and you can only reify parameterized types in inline functions

**Pull request merged**